arbnode/blob_reader.go

```diff
@@ -31,29 +32,45 @@ type BlobClient struct {
 	// The genesis time time and seconds per slot won't change so only request them once.
 	cachedGenesisTime    uint64
 	cachedSecondsPerSlot uint64
+
+	// Directory to save the fetcehd blobs
+	blobDirectory string
 }
 
 type BlobClientConfig struct {
 	BeaconChainUrl string `koanf:"beacon-chain-url"`
+	BlobDirectory  string `koanf:"blob-directory"`
 }
 
 var DefaultBlobClientConfig = BlobClientConfig{
 	BeaconChainUrl: "",
+	BlobDirectory:  "",
 }
 
 func BlobClientAddOptions(prefix string, f *pflag.FlagSet) {
 	f.String(prefix+".beacon-chain-url", DefaultBlobClientConfig.BeaconChainUrl, "Beacon Chain url to use for fetching blobs")
+	f.String(prefix+".blob-directory", DefaultBlobClientConfig.BlobDirectory, "Full path of the directory to save fetched blobs")
 }
 
 func NewBlobClient(config BlobClientConfig, ec arbutil.L1Interface) (*BlobClient, error) {
 	beaconUrl, err := url.Parse(config.BeaconChainUrl)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse beacon chain URL: %w", err)
 	}
+	if _, err = os.Stat(config.BlobDirectory); err != nil {
```

Comment:
Do we want to have it be optional to have the blob directory? I think this will error with the default string "", but later you're checking `blobDirectory != ""` before saving to disk

Reply:
yep, we need it to be optional, fixed it.